includes/admin.form.inc

```diff
@@ -351,22 +354,1218 @@ function islandora_scholar_admin_pdf_form($form, &$form_state) {
+    '#states' => array(
+      'visible' => array(
+        ':input[name="islandora_scholar_xpaths_enabled"]' => array('checked' => TRUE),
+      ),
+    ),
+    '#ajax' => array(
+      'callback' => 'islandora_scholar_xpaths_citeproc_callback',
+      'wrapper' => 'islandora_scholar_xpaths_citeproc_title',
+    ),
+   '#field_suffix' => (!empty($form_state['values']['islandora_scholar_xpaths_demo']) ? _mods_id($form_state['values']['islandora_scholar_xpaths_local_container-author'], check_plain($form_state['values']['islandora_scholar_xpaths_demo'])) : t("Modify to refresh value (PID for demo object is required).")),
+  ));
+  $form['islandora_scholar_xpaths_citeproc'] += _islandora_scholar_add_variable('islandora_scholar_xpaths_local_collection-editor', '//mods:mods[1]/mods:relatedItem[@type="series"]/mods:name', array(
+    '#type' => 'textfield',
+    '#title' => t('Collection Editor'),
+    '#description' => check_plain('Default Value: //mods:mods[1]/mods:relatedItem[@type="series"]/mods:name <br/>XPath to use for the collection-editor.'),
+    '#states' => array(
+      'visible' => array(
+        ':input[name="islandora_scholar_xpaths_enabled"]' => array('checked' => TRUE),
+      ),
+    ),
+    '#ajax' => array(
+      'callback' => 'islandora_scholar_xpaths_citeproc_callback',
+      'wrapper' => 'islandora_scholar_xpaths_citeproc_title',
+    ),
+   '#field_suffix' => (!empty($form_state['values']['islandora_scholar_xpaths_demo']) ? _mods_id($form_state['values']['islandora_scholar_xpaths_local_collection-editor'], check_plain($form_state['values']['islandora_scholar_xpaths_demo'])) : t("Modify to refresh value (PID for demo object is required).")),
+  ));
+  $form['islandora_scholar_xpaths_citeproc'] += _islandora_scholar_add_variable('islandora_scholar_xpaths_role_term', 'mods:role/mods:roleTerm', array(
+    '#type' => 'textfield',
+    '#title' => t('Role Term'),
+    '#description' => check_plain('Default Value: mods:role/mods:roleTerm <br/>XPath to use for the role_term.'),
+    '#states' => array(
+      'visible' => array(
+        ':input[name="islandora_scholar_xpaths_enabled"]' => array('checked' => TRUE),
+      ),
+    ),
+    '#ajax' => array(
+      'callback' => 'islandora_scholar_xpaths_citeproc_callback',
+      'wrapper' => 'islandora_scholar_xpaths_citeproc_title',
+    ),
+   '#field_suffix' => (!empty($form_state['values']['islandora_scholar_xpaths_demo']) ? _mods_id($form_state['values']['islandora_scholar_xpaths_role_term'], check_plain($form_state['values']['islandora_scholar_xpaths_demo'])) : t("Modify to refresh value (PID for demo object is required).")),
+  ));
+  drupal_add_js("(function($) {
+    Drupal.behaviors.islandora_scholar = {
+      attach: function (context, settings) {
+      $('body').on('load', function(){
+        console.log('xxx');
+      });
+      }
+    };
+  })(jQuery);", 'inline');
```

Comment:
Am I seeing this right that [L1314](https://github.com/Islandora/islandora_scholar/pull/299/files#diff-0cdd19abe14e51fe9e7f061ec72c86d3R1314) through [L1322](https://github.com/Islandora/islandora_scholar/pull/299/files#diff-0cdd19abe14e51fe9e7f061ec72c86d3R1322) were added for debugging purposes only?

Reply:
Nope, that's actually for the refresh button. I didn't see a way to trigger a refresh of all of the groups without this. Using AJAX to refresh a $form['group_a'] is easy but having it also refresh $form['group_b'] didn't seem plausible.

Reply:
I see. Thank you @DonRichards for the clarification. I was lead to the assumption above by [L1318](https://github.com/Islandora/islandora_scholar/pull/299/files#diff-0cdd19abe14e51fe9e7f061ec72c86d3R1318) (`console.log('xxx');`) - should that line be removed in order not to clutter up the user's browser-log?